### PR TITLE
feat: integrate dockerised prometheus and grafana

### PR DESCRIPTION
**Pull request by burnerlee**
this PR adds dockerised Prometheus and Grafana to evalai. The Prometheus server starts at 9090, and the grafana server at 3000. The password for admin user on Grafana is `evalai`. Prometheus has been added as a data source on grafana

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -64,3 +64,29 @@ services:
...
+      - '--config.file=/etc/prometheus/prometheus.yml'
+    restart: unless-stopped
+    ports:
+      - '9090:9090'
+  
+  grafana:
+    image: grafana/grafana:latest
+    user: "1000"
+    environment:
+      - GF_SECURITY_ADMIN_PASSWORD=evalai
```
Comment (Ram81) [inline docker-compose.yml:86]: @burnerlee move this to a `docker.env` file for grafana. We can't check in the secrets in the repository

File: .gitignore [hunk 2]
```diff
@@ -67,3 +67,5 @@ ssl/
 out/
 .idea_modules/
 
+#monitoring-setup
```
Comment (Ram81) [inline .gitignore:70]: Please add a space after `#`

File: prometheus/prometheus.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+global:
+  scrape_interval: 5s
```
Comment (Ram81) [inline prometheus/prometheus.yml:2]: @burnerlee we can increase this to 30s

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 repos:
 -   repo: https://github.com/ambv/black
-    rev: stable
+    rev: 21.5b1
```
Comment (Ram81) [inline .pre-commit-config.yaml:3]: @burnerlee you can remove this change from the PR

File: .gitignore [hunk 5]
```diff
@@ -67,3 +67,5 @@ ssl/
...
+prometheus/prometheus_db/
```
Comment (Ram81) [inline .gitignore:71]: Please add `grafana_db` in gitignore too.

**Review by Ram81 — CHANGES_REQUESTED** [review]

File: docker/dev/docker.env [hunk 6]
```diff
@@ -32,3 +32,5 @@ EKS_NODEGROUP_ROLE_ARN=x
 C_FORCE_ROOT=true
 ENVIRONMENT=dev
 SERVICE_DNS=localhost
+
+GF_SECURITY_ADMIN_PASSWORD=evalai
```
Comment (Ram81) [inline docker/dev/docker.env:36]: @burnerlee can you also add environment variables to configure username

File: .gitignore [hunk 7]
```diff
@@ -67,3 +67,7 @@ ssl/
 out/
 .idea_modules/
 
+# Monitoring-Setup
+prometheus/prometheus_db/*
+!prometheus/prometheus_db/.gitkeep
```
Comment (Ram81) [inline .gitignore:72]: @burnerlee the `.gitkeep` files are not being ignored. Please fix that too.

File: .gitignore [hunk 8]
```diff
@@ -67,3 +67,7 @@ ssl/
 out/
 .idea_modules/
 
+# Monitoring-Setup
```
Comment (Ram81) [inline .gitignore:70]: remove `-`

**Review by Ram81 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 7 above)
Comment (burnerlee) [reply .gitignore:72]: @Ram81 I committed these files to push the grafana/grafana_db, prometheus/prometheus_db folders, as these directories are currently empty, i need to keep a gitkeep file in them and commit that file

Comment (Ram81) [reply .gitignore:72]: @burnerlee we don't need to check in the folders if there is nothing in that folder. And you can remove `.gitkeep` from `promethues` folder as we already have the config file in that folder.

**Review by Ram81 — CHANGES_REQUESTED** [review]
@burnerlee let's create a single `monitoring` folder and move all grfana and prometheus config files in that folder. Please make the required changes to the docker setup

File: docker/dev/docker.env [hunk 9]
```diff
@@ -32,3 +32,6 @@ EKS_NODEGROUP_ROLE_ARN=x
 C_FORCE_ROOT=true
 ENVIRONMENT=dev
 SERVICE_DNS=localhost
+
+GF_SECURITY_ADMIN_USER=evalai
+GF_SECURITY_ADMIN_PASSWORD=evalaiai
```
Comment (Ram81) [inline docker/dev/docker.env:37]: @burnerlee can we change the username to `admin` and password to `password` for dev

**Review by Ram81 — CHANGES_REQUESTED** [review]

**Review by Ram81 — APPROVED** [review]
LGTM. @RishabhJain2018 can you review the PR

**Review by RishabhJain2018 — APPROVED** [review]
This looks good!

**Pull request merged**